src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIRST_MAX_NUMBER = 5;
+    private static final int SECOND_MAX_NUMBER = 100;
+    private static final int RADIUS = 10;
+    private final Random random;
+    private final ColorSupplier colorSupplier;
+
+    public FigureSupplier(Random random, ColorSupplier colorSupplier) {
+        this.random = random;
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Figure getRandomFigure() {
+        int max1 = random.nextInt(SECOND_MAX_NUMBER) + 1;
+        int max2 = random.nextInt(SECOND_MAX_NUMBER) + 1;
+        Figure figure = switch (random.nextInt(FIRST_MAX_NUMBER) + 1) {
+            case 1 -> new Rectangle(max1, max2);
+            case 2 -> new RightTriangle(max1, max2);
+            case 3 -> new Ellipse(max1, max2);
+            case 4 -> new Square(max1);
+            case 5 -> new IsoscelesTrapezoid(max1, max2, max2);
+            default -> throw new IllegalArgumentException("Unexpected value");

```

Comment:
The `default` case in the switch statement is unnecessary because the random number generation is controlled by `FIRST_MAX_NUMBER` and should not exceed this range. Consider removing the `default` case or handling it in a way that aligns with the expected behavior.